actions/setup-argo/action.yaml

```diff
@@ -0,0 +1,62 @@
+name: Install Argo cli
+description: Install Argo cli
+
+inputs:
+  version:
+    description: |
+      Version of the Argo CLI to install.
+    default: 3.5.1
+
+runs:
+  using: composite
+
+  steps:
+    - name: Setup cache
+      id: cache
+      uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+      with:
+        path: ${{ github.workspace }}/bin/argo
+        key: argo-${{ runner.os }}-${{ runner.arch }}-${{ inputs.version }}
+
+    - name: Normalize runner.os to match Argo release
+      shell: sh
+      env:
+        OS: ${{ runner.os }}
+      run: |
+        OS=$(echo "$OS" | tr '[:upper:]' '[:lower:]')
```

Comment:
@iainlane I can't do the same as on https://github.com/grafana/shared-workflows/pull/216 because I also need to lowercase it for argo.

Reply:
Setting up both the variables in one step with a couple of `case` statements might be the most straightforward

Reply:
What's wrong with the current script? First lowercase it, then `s/macos/darwin`.

Reply:
Merging so I can keep testing it, we can iterate on the finer details in a follow up.

Reply:
I find it a bit convoluted doing all these steps when you could directly map the values